### PR TITLE
Halstack upgrade

### PR DESCRIPTION
**Pull request by marcialfps**
**Purpose**
This library should work correctly with Halstack React version 6 and its peer dependencies.

**Description**
To complete the PR purpose I made the following changes:
- Upgrade dependencies in `/lib` (React, Halstack React, and Halstack Client) and remove unused `axios` dependency.
- Use of new Text input component in HalAutocomplete.
- Fix HalAutocomplete filtering. Now, this is the responsibility of HAL API.
- Code refactor.
- Add HalTable and HalAutocomplete examples to `/app`.
- Upgrade dependencies and remove the ones unused from `/app`.
- Upgrade and improvements to the documentation from the README file.

**Screenshots**
image
image

image
image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -72,87 +68,70 @@ export default () => {
...
 | :-------------------------------------- | :------ | :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
 | url: `string`                           |         | The URL of the resource. `Required`                                                                                                                                                                                                                                                             |
 | headers: `Object`                       |         | Contains the http headers to be sent along with the http requests to the url indicated in the `url` prop. `Optional`                                                                                                                                                                            |
```
Comment (Lisa159) [inline README.md:120]: As `url` is an abbreviation, it should be written in uppercase `URL`.

File: README.md [hunk 2]
```diff
@@ -72,87 +68,70 @@ export default () => {
...
 | resourceInteractions: `Object`                                                             | This is an object containing as many entries as interactions (\_options.links) are available in the HAL resource. Each entry has the rel of the interaction as a key, and is a function that you can execute passing a payload as a parameter. Executing one of these functions will: <ul><li>Make the http request associated to the given interaction.</li><li>Change the `requestStatus` to `'interaction'`, and then back to `'resolved'` (even when the request fails).</li><li>Update the `resource` if the request responds with a new representation of the same resource.</li><li>Return a promise, so that you can handle the resolution or rejection manually.</li></ul> |
```
Comment (Lisa159) [inline README.md:132]: What means the word `rel`?

Comment (Lisa159) [reply README.md:132]: Also, I think would be better to put `_options.links` between ``.

File: README.md [hunk 3]
```diff
@@ -31,23 +31,19 @@ Hooks
...
 | headers: `Object`                       |         | Contains the http headers to be sent along with the http requests to the collectionUrl. `Optional`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |
```
[the remark this answers is not in the document]
Comment (marcialfps) [reply README.md:39]: Done.

File: README.md (hunk 1 above)
Comment (marcialfps) [reply README.md:120]: Done.

File: README.md (hunk 2 above)
Comment (marcialfps) [reply README.md]: Done!

File: lib/src/components/HalTable.jsx [hunk 4]
```diff
@@ -109,24 +106,26 @@ const HalTable = ({ collectionUrl, asyncHeadersHandler, headers, columns, itemsP
     <DxcHALTableContainer>
```
Comment (Lisa159) [inline lib/src/components/HalTable.jsx:106]: Maybe to rename this `DxcHALTableContainer`? We are not using the word `HAL` in uppercase anymore and I would keep the word `Dxc` for the components of `Halstack React`. Something like `HalTableContainer` I think would be better.

File: lib/src/components/HalTable.jsx [hunk 5]
```diff
@@ -109,24 +106,26 @@ const HalTable = ({ collectionUrl, asyncHeadersHandler, headers, columns, itemsP
...
       <DxcTable>
         <HeaderRow>
```
Comment (Lisa159) [inline lib/src/components/HalTable.jsx:108]: Why this `HeaderRow` has a height of `60px`? Is it in the desgin?

Comment (marcialfps) [reply lib/src/components/HalTable.jsx:108]: I think it uses the same design as the [Resultset Table](https://developer.dxc.com/halstack/next/components/resultset-table/specifications/).

Comment (Lisa159) [reply lib/src/components/HalTable.jsx:108]: So, in that case, there are some values that are not matching the design. 
- Height of the cells: as it is specified, it should be 60px but in the component is less.

Component:
image

Design:
image

- Paddings in the cells: the last cell of the row should have a `padding-right: 40px`.

Component:
image

Design:
image

- I would put tab index for the column name when it has sorting. Also some space between label and icon.

Component:
image

Design:
image

File: lib/src/components/HalTable.jsx (hunk 4 above)
Comment (marcialfps) [reply lib/src/components/HalTable.jsx]: I agree. Done!

**Review by Lisa159 — APPROVED** [review]

**Pull request merged**